tonic/src/codec/compression.rs

```diff
@@ -160,29 +208,23 @@ impl CompressionEncoding {
     }
 
     #[allow(missing_docs)]
-    #[cfg(any(feature = "gzip", feature = "zstd"))]
     pub(crate) fn as_str(&self) -> &'static str {
+        #[cfg(any(feature = "gzip", feature = "zstd"))]
         match self {
             #[cfg(feature = "gzip")]
             CompressionEncoding::Gzip => "gzip",
             #[cfg(feature = "zstd")]
             CompressionEncoding::Zstd => "zstd",
         }
+
+        #[cfg(not(any(feature = "gzip", feature = "zstd")))]
+        unreachable!("CompressionEncoding::as_str called without any compression features enabled")
```

Comment:
This seems problematic? Why do this?

Reply:
because rust won't let you have a `match` on an empty enum (which is what happens when both are turned off)